### PR TITLE
packed version of abi.encode of auction data

### PR DESCRIPTION
**Pull request by josojo**
as title

---
testplan: unit tests

### CODE REVIEW

File: test/stablecoin_converter.js [hunk 1]
```diff
@@ -1249,24 +1249,23 @@ contract("StablecoinConverter", async (accounts) => {
...
-      sellTokenBalance: parseInt(element.slice(1 * HEX_WORD_SIZE, 2 * HEX_WORD_SIZE), 16),
-      buyToken: parseInt(element.slice(2 * HEX_WORD_SIZE, 3 * HEX_WORD_SIZE), 16),
-      sellToken: parseInt(element.slice(3 * HEX_WORD_SIZE, 4 * HEX_WORD_SIZE), 16),
-      validFrom: parseInt(element.slice(4 * HEX_WORD_SIZE, 5 * HEX_WORD_SIZE), 16),
-      validUntil: parseInt(element.slice(5 * HEX_WORD_SIZE, 6 * HEX_WORD_SIZE), 16),
-      isSellOrder: parseInt(element.slice(6 * HEX_WORD_SIZE, 7 * HEX_WORD_SIZE), 16) > 0,
-      priceNumerator: parseInt(element.slice(7 * HEX_WORD_SIZE, 8 * HEX_WORD_SIZE), 16),
-      priceDenominator: parseInt(element.slice(8 * HEX_WORD_SIZE, 9 * HEX_WORD_SIZE), 16),
-      remainingAmount: parseInt(element.slice(9 * HEX_WORD_SIZE, 10 * HEX_WORD_SIZE), 16),
+      user: "0x" + element.slice(0, 20 * 2), // address is only 20 bytes
```
Comment (fleupold) [inline test/stablecoin_converter.js]: I'm not sure if all this arithmetic below here really helps comprehension (it confuses me). I'd just use the final numbers.

**Review by fleupold — APPROVED** [review]
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia0.giphy.com%2Fmedia%2F8vR91bniSOQMXCNJmF%2Fgiphy.gif&token=packed" data-gitmeme-token="packed"></a>

File: test/stablecoin_converter.js (hunk 1 above)
Comment (josojo) [reply test/stablecoin_converter.js]: I think it helps a lot as the same numbers also appear in the driver-code in rust.

Comment (bh2smith) [reply test/stablecoin_converter.js]: cute!

**Review by bh2smith — APPROVED** [review]

File: test/stablecoin_converter.js (hunk 1 above)
Comment (fleupold) [reply test/stablecoin_converter.js]: Hmh, maybe a `pop` logic similar to what we are doing in the driver is easier to read?

Like:
```js
const user = parseInt(bytes.splice(0, ADDRESS_WIDTH))
const sellTokenBalance = parseInt(bytes.splice(0, UINT256_WIDTH))
...
```

Comment (bh2smith) [reply test/stablecoin_converter.js]: I like that I can see the indices like in the `from_bytes` in the driver code, but I do agree that it looks a bit repetitive to the point that I wouldn't read it anyway. Perhaps the most recent suggestion is best.

File: contracts/StablecoinConverter.sol [hunk 2]
```diff
@@ -371,16 +371,16 @@ contract StablecoinConverter is EpochTokenLocker {
...
+        element = abi.encodePacked(user);
+        element = element.concat(abi.encodePacked(sellTokenBalance));
+        element = element.concat(abi.encodePacked(order.buyToken));
+        element = element.concat(abi.encodePacked(order.sellToken));
+        element = element.concat(abi.encodePacked(order.validFrom));
+        element = element.concat(abi.encodePacked(order.validUntil));
+        element = element.concat(abi.encodePacked(order.isSellOrder));
+        element = element.concat(abi.encodePacked(order.priceNumerator));
+        element = element.concat(abi.encodePacked(order.priceDenominator));
+        element = element.concat(abi.encodePacked(order.remainingAmount));
```
Comment (anxolin) [inline contracts/StablecoinConverter.sol:383]: @josojo  Maybe is a silly question, but can't we just encode all in one call to `encodePacked`:

`abi.encodePacked(user, sellTokenBalance, order.buyToken, .... )`

**Review by anxolin — COMMENTED** [review]
Looks more packed now :)

File: contracts/StablecoinConverter.sol (hunk 2 above)
Comment (josojo) [reply contracts/StablecoinConverter.sol:383]: Yeah, for sure. But i like it better in the current version, as it is easier to read.

Comment (anxolin) [reply contracts/StablecoinConverter.sol:383]: Aha, ok, so it's equivalente? The compiler will turn that into the same?

Why do you think it's easier to read? The other version doesn't need to be a one-liner:
```solidity
abi.encodePacked(
    user,
    sellTokenBalance,
    order.buyToken,
    ....
)
```

**Pull request merged**